Bottom: d345ffc9c640135ca0de6638a27aeb8fcdaf72f9
Top:    ad01e2a21366fd53c049909d2943604bae7081b4
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-05 11:10:33 +0800

Refresh of exp

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index d97d695..4c31c1c 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1589,13 +1589,8 @@ static void tun_rx_batched(struct tun_struct *tun, struct tun_file *tfile,
 	spin_unlock(&queue->lock);
 
 	if (rcv) {
-		struct sk_buff *nskb, *tmp;
-
 		local_bh_disable();
-		list_for_each_entry_safe(nskb, tmp, &list, list) {
-			list_del_init(&nskb->list);
-			kfree_skb(nskb);
-		}
+		netif_receive_skb_list(&list);
 		local_bh_enable();
 	}
 }
